### PR TITLE
IRGen: getSingletonAggregateFieldType must not return field if its ac…

### PR DESCRIPTION
**Pull request by aschwaighofer**
…cess level does not match

rdar://50554717

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by aschwaighofer** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
Nice catch!

File: lib/IRGen/GenType.cpp [hunk 1]
```diff
@@ -2289,8 +2289,17 @@ SILType irgen::getSingletonAggregateFieldType(IRGenModule &IGM, SILType t,
     auto allFields = structDecl->getStoredProperties();
     
     auto field = allFields.begin();
-    if (!allFields.empty() && std::next(field) == allFields.end())
-      return t.getFieldType(*field, IGM.getSILModule());
+    if (!allFields.empty() && std::next(field) == allFields.end()) {
+      auto fieldTy = t.getFieldType(*field, IGM.getSILModule());
+      if (auto fieldDecl = fieldTy.getNominalOrBoundGenericNominal()) {
```
Comment (slavapestov) [inline lib/IRGen/GenType.cpp]: What if the field type is a tuple containing an inaccessible type? I think SIL Type lowering has a notion of an “ABI visible” type that handles such cases, and we just need to use the right API here

Comment (aschwaighofer) [reply lib/IRGen/GenType.cpp]: Tuples, tuples, tuples!

https://github.com/apple/swift/pull/25259

**Pull request merged**